File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.java
class Solution {
    public int solution(int a, int b, int n) {
        int answer = 0;
        
        while (a <= n) {
            int exchange = n / a;
            answer += exchange * b;
            n = (n % a) + (exchange * b);
        }
        
        return answer;
    }
}

File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.kt
class Solution {
    fun solution(a: Int, b: Int, n: Int): Int {
        var answer = 0
        var cur = n
        
        while (a <= cur) {
            val exchange = cur / a
            answer += exchange * b
            cur = (cur % a) + (exchange * b)
        }
        
        return answer
    }
}